### ISSUE TITLE
Pyright allows nested TypedDict definition

### ISSUE THREAD

**Issue by betafcc** [microsoft/pyright#4016]
Don't really look like a bug (please don't get rid of it 😅), but worth pointing

This works as expected in pyright:

```python
from typing import TypedDict

Geometry = TypedDict(
    "Geometry",
    {
        "position": TypedDict("Position", {"x": int, "y": int, "z": int}),
        "dimensions": TypedDict("Dimensions", {"width": int, "height": int}),
    },
)


def area(geometry: Geometry):
    return geometry["dimensions"]["width"] * geometry["dimensions"]["height"]


r = area({"position": {"x": 0, "y": 0, "z": 0}, "dimensions": {"width": 10, "height": 20}})

reveal_type(r)  # int as expected
```

And I though was standard python hinting feature but just noticed it breaks on mypy:

```
helpers/di.py:6: error: Inline TypedDict types not supported; use assignment to define TypedDict
helpers/di.py:13: error: TypedDict "TypedDict" has no key "dimensions"
helpers/di.py:16: error: Extra keys ("position", "dimensions") for TypedDict "TypedDict"
helpers/di.py:17: note: Revealed type is "Any"
Found 3 errors in 1 file (checked 1 source file)
```

And is referenced in: https://github.com/python/mypy/issues/5149, since it seems to be some discussion if this feature will be officially implemented or what format will have, and didn't find mention of it in pyright docs, seems to not be intended.

**Comment by erictraut** [comment]
This isn't a bug. Call expressions are expressly forbidden in a type annotation like you have done here.

**Comment by JelleZijlstra** [comment]
@erictraut the bug (which I confirmed) is that pyright allows the code in the OP without errors.

**Comment by erictraut** [comment]
Ah yes, this is a false negative. Pyright should emit an error in this case.

**Comment by betafcc** [comment]
> This isn't a bug. Call expressions are expressly forbidden in a type annotation like you have done here.

yeah sorry, I ended up phrasing in a confusing way, cause it is a bug but a bug I wish was a feature 🥲

thanks for the quick response

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.275, which I just published. It will also be included in a future release of pylance.
